model/src/main/java/hu/blackbelt/judo/meta/jsl/scoping/JslDslIndex.xtend

```diff
@@ -46,19 +90,32 @@ class JslDslIndex {
 
 
 	def getImportedEntityDeclarations(ModelDeclaration model, EClass instance) {
-		val importNames = model.imports.map[i | i.modelName.importName].toList
+		val importNames = model.imports.map[i | i.model.name].toList
 		model.getVisibleEObjectDescriptions.filter[d |
 			importNames.exists[i | d.qualifiedName.toString.startsWith(i) && instance.isInstance(d.EObjectOrProxy)]
 		].toList
 	}
 
+	def getImportedModelDeclarationFullyQualifiedName(ModelImportDeclaration model) {
+		
+		getVisibleEObjectDescriptions(JsldslPackage::eINSTANCE.modelImportDeclaration)
+		
+		//val importNames = model.imports.map[i | i.model.name].toList
+		//model.getVisibleEObjectDescriptions.filter[d |
+		//	importNames.exists[i | d.qualifiedName.toString.startsWith(i) && instance.isInstance(d.EObjectOrProxy)]
+		//].toList
+	}
+
 	def getVisibleContainers(EObject o) {
+		if (o === null || o.eResource === null) {
```

Comment:
=== is okay here, but not elsewhere?

Reply:
I've replaced everywhere and tests are OK.